client/src/components/Notifications.js

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect, useContext, useReducer } from "react";
+import socket from "../services/sockets";
+import axios from "axios";
+import userContext from "../userContext";
+//MaterialUI imports
+import { makeStyles } from "@material-ui/core";
+import Badge from "@material-ui/core/Badge";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
+import Grid from "@material-ui/core/Grid";
+import IconButton from "@material-ui/core/IconButton";
+import Typography from "@material-ui/core/Typography";
+import NotificationsNoneIcon from "@material-ui/icons/NotificationsNone";
+
+const useStyles = makeStyles({
+  menu: {
+    padding: "0",
+  },
+  seen: { background: "white" },
+  unseen: {
+    background: "blue",
+  },
+  wrapper: {
+    padding: "0 1vh 0 0",
+    margin: "0",
+  },
+  time: {
+    color: "gray",
+  },
+});
+
+const initialState = {
+  notifications: [],
+};
+
+const reducer = (state, action) => {
+  const updateUnread = () => {
+    const unread = state.notifications.map((item) => {
+      item.seen = true;
+      return item;
+    });
+    axios.put(`/notifications/update-read`, { notifications: unread });
+    return unread;
+  };
+
+  switch (action.type) {
+    case "newNotification":
+      return { notifications: [action.payload, ...state.notifications] };
+    case "getNotifications":
+      return { notifications: action.payload };
+    case "setSeen":
+      return { notifications: updateUnread() };
+    default:
+      throw new Error("State update error");
+  }
+};
```

Comment:
could put this logic out of the component. Can also do the same with the API call